lib/route.ts

```diff
@@ -7,24 +7,65 @@ import {
 
 import { createContext, type Operation } from "./deps/effection.ts";
 
-const ParamsContext = createContext<MatchResult["params"]>(
-  "revolution.params",
-);
+const ParamsContext = createContext<MatchResult["params"]>("revolution.params");
 
 export function* useParams<T extends object>(): Operation<T> {
   return (yield* ParamsContext) as T;
 }
 
+function buildMethodRoute<T>(
+  method: "GET" | "POST" | "PUT" | "DELETE" | "PATCH",
+  handler: Handler<Request, T>,
+): Middleware<Request, T> {
+  return function* (request, next) {
+    if (request.method === method) {
+      return yield* handler(request);
+    } else {
+      return yield* next(request);
+    }
+  };
+}
+
+export function GET<T>(handler: Handler<Request, T>): Middleware<Request, T> {
+  return buildMethodRoute("GET", handler);
+}
+
+export function POST<T>(handler: Handler<Request, T>): Middleware<Request, T> {
+  return buildMethodRoute("POST", handler);
+}
+
+export function PUT<T>(handler: Handler<Request, T>): Middleware<Request, T> {
+  return buildMethodRoute("PUT", handler);
+}
+
+export function PATCH<T>(handler: Handler<Request, T>): Middleware<Request, T> {
+  return buildMethodRoute("PATCH", handler);
+}
+
+export function DELETE<T>(
+  handler: Handler<Request, T>,
+): Middleware<Request, T> {
+  return buildMethodRoute("DELETE", handler);
+}
+
 export function route<T>(
   path: string,
-  handler: Middleware<Request, T>,
+  ...middlewares: Middleware<Request, T>[]
 ): Middleware<Request, T> {
+  const inlinedMiddleware = middlewares.slice(1).reduce(
+    (inlined, current) => {
+      return function* (request, next) {
+        return yield* inlined(request, (request) => current(request, next));
+      };
+    },
+    middlewares[0],
+  );
```

Comment:
I think we can just use the `concat()` helper which already collapses a single middleware out of one.